mono/metadata/handle.h

```diff
@@ -538,6 +523,8 @@ mono_string_handle_pin_chars (MonoStringHandle s, uint32_t *gchandle_out);
 gpointer
 mono_object_handle_pin_unbox (MonoObjectHandle boxed_valuetype_obj, uint32_t *gchandle_out);
 
+#define mono_handle_unbox(h) (mono_object_unbox (MONO_HANDLE_RAW (h)))
```

Comment:
This returns a raw pointer into the middle of a managed object that is kept alive by a `MonoObjectHandle` (ie, potentially movable). We really don't want that to be possible.

Reply:
My mistake. It wasn't pinned before.

Reply:
Should this be renamed and rewritten mono_handle_pin_and_unbox?

Reply:
Or maybe the pinning is just gone now anyway, I'm triple checking.

Reply:
The unboxing is there, like the pointer-using version. I'll fix the handle one to pin and unbox.

Reply:
Isn't this one of those things, you had pointed out, about passing right to managed, managed keeps it alive? And what about the non-unbox path? Pin both?

Reply:
Every mono_runtime_invoke_checked needs work..